src/main/java/org/kuali/kfs/module/purap/document/service/impl/PaymentRequestServiceImpl.java

```diff
@@ -0,0 +1,2108 @@
+		// limit is used.
+		if (ObjectUtils.isNull(minimumAmount) || defaultMinimumLimit.compareTo(minimumAmount) < 0) {
+			minimumAmount = defaultMinimumLimit;
+		}
+
+		// The document is eligible for auto-approval if the document total is below the
+		// limit.
+		if (document.getFinancialSystemDocumentHeader().getFinancialDocumentTotalAmount().isLessThan(minimumAmount)) {
+			if (LOG.isInfoEnabled()) {
+				LOG.info(" -- PayReq [" + document.getDocumentNumber() + "] auto-approved due to document Total ["
+						+ document.getFinancialSystemDocumentHeader().getFinancialDocumentTotalAmount()
+						+ "] being less than "
+						+ (minimumAmount == defaultMinimumLimit ? "Default Auto-Approval Limit "
+								: "Configured Auto-Approval Limit ")
+						+ "of " + (minimumAmount == null ? "null" : minimumAmount.toString()) + ".");
+			}
+			return true;
+		}
+
+		if (LOG.isInfoEnabled()) {
+			LOG.info(" -- PayReq [" + document.getDocumentNumber() + "] skipped due to document Total ["
+					+ document.getFinancialSystemDocumentHeader().getFinancialDocumentTotalAmount()
+					+ "] being greater than "
+					+ (minimumAmount == defaultMinimumLimit ? "Default Auto-Approval Limit "
+							: "Configured Auto-Approval Limit ")
+					+ "of " + (minimumAmount == null ? "null" : minimumAmount.toString()) + ".");
+		}
+
+		return false;
+	}
+
+	/**
+	 * This method iterates a collection of negative payment request approval limits
+	 * and returns the minimum of a given minimum amount and the least among the
+	 * limits in the collection.
+	 *
+	 * @param limits        The collection of NegativePaymentRequestApprovalLimit to
+	 *                      be used in determining the minimum limit amount.
+	 * @param minimumAmount The amount to be compared with the collection of
+	 *                      NegativePaymentRequestApprovalLimit to determine the
+	 *                      minimum limit amount.
+	 * @return The minimum of the given minimum amount and the least among the
+	 *         limits in the collection.
+	 */
+	protected KualiDecimal getMinimumLimitAmount(Collection<NegativePaymentRequestApprovalLimit> limits,
+			KualiDecimal minimumAmount) {
+		for (NegativePaymentRequestApprovalLimit limit : limits) {
+			KualiDecimal amount = limit.getNegativePaymentRequestApprovalLimitAmount();
+			if (null == minimumAmount) {
+				minimumAmount = amount;
+			} else if (minimumAmount.isGreaterThan(amount)) {
+				minimumAmount = amount;
+			}
+		}
+		return minimumAmount;
+	}
+
+	/**
+	 * Retrieves a list of payment request documents with the given vendor id and
+	 * invoice number.
+	 *
+	 * @param vendorHeaderGeneratedId The vendor header generated id.
+	 * @param vendorDetailAssignedId  The vendor detail assigned id.
+	 * @return List of payment request documents.
+	 */
+	/*
+	 * CU customization: backport FINP-8270
+	 */
+	@Override
+	public List<PaymentRequestView> getPaymentRequestsByVendorNumber(Integer vendorHeaderGeneratedId,
```

Comment:
It seemed strange that this method's return type was altered without also changing the PaymentRequestService interface (and likewise with KualiCo's changes), but then I saw that the interface declares this method with a raw return type instead of one using generics.

We may want to notify KualiCo about the raw-type issue on the PaymentRequestService interface if they're not aware of it.

Reply:
Good point Chad, I will notify KualiCo. Thank you.